compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11465,43 +11465,45 @@ OMR::Z::TreeEvaluator::loadaddrEvaluator(TR::Node * node, TR::CodeGenerator * cg
    if (symRef->isUnresolved())
       {
 #ifdef J9_PROJECT_SPECIFIC
-      if (cg->getS390ProcessorInfo()->supportsArch(TR_S390ProcessorInfo::TR_z10))
-         {
-         TR::UnresolvedDataSnippet * uds;
-         TR::Instruction * cursor;
+      // create UnresolvedDataSnippet
+      TR::UnresolvedDataSnippet * uds = new (cg->trHeapMemory()) TR::UnresolvedDataSnippet(cg, node, symRef, false, false);
+      cg->addSnippet(uds);
 
-         // create UnresolvedDataSnippet
-         uds = new (cg->trHeapMemory()) TR::UnresolvedDataSnippet(cg, node, symRef, false, false);
-         cg->addSnippet(uds);
+      // generate branch to the unresolved data snippet
+      TR::Instruction * cursor = generateRegUnresolvedSym(cg, TR::InstOpCode::getLoadOpCode(), node, targetRegister, symRef, uds);
+      uds->setBranchInstruction(cursor);
 
-         // generate branch to the unresolved data snippet
-         cursor = generateRegUnresolvedSym(cg, TR::InstOpCode::getLoadOpCode(), node, targetRegister, symRef, uds);
+      // create a patchable data in litpool
+      TR::S390WritableDataSnippet * litpool = cg->CreateWritableConstant(node);
+      litpool->setUnresolvedDataSnippet(uds);
 
-         uds->setBranchInstruction(cursor);
-
-         // create a patchable data in litpool
-         TR::S390WritableDataSnippet * litpool = cg->CreateWritableConstant(node);
-         litpool->setUnresolvedDataSnippet(uds);
-
-         TR::S390RILInstruction * LRLinst;
-         LRLinst = (TR::S390RILInstruction *) generateRILInstruction(cg, TR::InstOpCode::getLoadRelativeLongOpCode(), node, targetRegister, reinterpret_cast<void*>(0xBABE), 0);
+      TR::S390RILInstruction * LRLinst;
```

Comment:
I'd prefer if we initialize these pointers to `NULL` so we catch errors fast in case we don't initialize on some paths. Obviously for simple code such as this it we know it is initialized however it's good practice to initialize such pointers everywhere (for more complicated code paths where it is not obvious the variable is initialized on all paths).

Reply:
Variable initialized to NULL.